src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -512,12 +532,16 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
         [RequiresUnreferencedCode("Cannot statically analyze what the element type is of the object collection so its members may be trimmed.")]
         private static IEnumerable BindExistingCollection(IEnumerable source, IConfiguration config, BinderOptions options)
         {
-            // find the interface that is IEnumerable<T>
-            Type type = source.GetType().GetInterface("IEnumerable`1", false)!;
-            Type elementType = type.GenericTypeArguments[0];
-            Type genericType = typeof(List<>).MakeGenericType(elementType);
+            var (collectionType, elementType) = ResolveSuitableCollectionFromSourceCollection(source);
```

Comment:
Can we add some assert here that we never get non-generic IEnumerable interface. Just to protect against any wrong future use of this method.

Reply:
Good idea, but that method has gone since the rebase from main